lib/atc/aws/fixity_check.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class Atc::Aws::FixityCheck
+  def initialize(stored_object, stream_id)
+    @bucket_name = stored_object.storage_provider.container_name
+    @object_path = stored_object.path
+    @fixity_checksum_algorithm = stored_object.source_object.fixity_checksum_algorithm
+    @stream_id = stream_id
```

Comment:
Maybe it would be simpler to remove `stream_id` param and instead assign stored_object.id to @stream_id?